src/commands/android/execute-commands.ts

```diff
@@ -0,0 +1,82 @@
+import path from "path";
+import colors from "ansi-colors";
+import * as dotenv from "dotenv";
+
+import Logger from "../../logger";
+import { getPlatformName } from "../../utils";
+import { Options, Platform } from "./interfaces";
+import { AVAILABLE_SUBCOMMANDS } from "./constants";
+import { getSdkRootFromEnv, showHelp } from "./utils/common";
+import { connectWirelessAdb } from "./commands/connect/connect-wireless-adb";
+
+export class SdkCommandExecute {
+  sdkRoot: string;
+  options: Options;
+  subcommand: string;
+  rootDir: string;
+  platform: Platform;
+  androidHomeInGlobalEnv: boolean;
+
+  constructor(subcommand: string, options: Options, rootDir = process.cwd()) {
+    this.sdkRoot = "";
+    this.options = options;
+    this.subcommand = subcommand;
+    this.rootDir = rootDir;
+    this.platform = getPlatformName();
+    this.androidHomeInGlobalEnv = false;
+  }
+
+  async run(): Promise<boolean> {
+    if (!Object.keys(AVAILABLE_SUBCOMMANDS).includes(this.subcommand)) {
+      showHelp([], this.subcommand);
```

Comment:
Maybe do a separate implementation of `showHelp` for subcommands? The help content for the subcommands is anyways an extension to the original help content and can be pulled out as a separate function. We don't need to force-implement everything into a single function if splitting the implementation into different functions makes more sense.

The main motive here is to show help at every step to the users that makes sense to them. So, the help section at every different level of the project would be different.

For ex., if you run `npx @nightwatch/mobile-helper andr`, it will show you a very minimal help section with all the commands supported. And if you go one level inside and make some mistake in entering the options for the `android` command (instead of the `android` command itself), it will show you help related to the options. 

But since we are integrating the subcommands with the `android` command itself, the latter case should also show all the available subcommands along with all the available options (in case user actually wanted to run a subcommand), as you correctly did in the other PR. But instead of merging everything together in a single function, we can split them out into two (or more) functions, each handling a specific section of the help section. And the division of these sections would be in terms of usage:

* Usage 1: Verify if all the requirements are met to run tests on an Android device/emulator.
  > Display all available options to verify/setup.
* Usage 2: Control and manage your already existing Android SDK setup using subcommands.
  > Display all available subcommands with major options (not all of them).

Now, once this is done, we might need to require a separate Help for each sub-commands, showing the options for that one subcommand only. This is for the case when the user got the subcommand right but messed up with the options. The implementation could be done in the `android/subcommands/<command-name>/index.ts` file. 

But remember not to duplicate any data for these individual cases. For ex., the data for all the available subcommands and their corresponding options should come from a single place or constant, and we'll use that data differently in different help section implementations.

Reply:
Please let me know if there's anything that don't make sense here.

Reply:
@garg3133 Thanks. I understood it pretty well. As for dividing the `showHelp` in seperate functions, will it be right to keep the current `showHelp` in `AndroidSetup` as it is and create separate function for subcommands help and invoke it inside both current `showHelp` function and inside the `AndroidSubcommand` class? Its because the current `showHelp` is not required anywhere else and will be always called in just one place, that is inside `AndroidSetup`.

For eg. 
Lets create `showSubcommandHelp` and put it inside `common.ts`.
1. If user passes `npx @nightwatch/mobile-helper android -h` then `showHelp` inside `AndroidSetup` will execute and display all options. Then it will invoke `showSubcommandHelp` and display help for available subcommands.
2. If user passes `npx @nightwatch/mobile-helper android random` then only `showSubcommandHelp` will be called inside `AndroidSubcommand` class.

Reply:
Makes sense.